core/dbt/config/runtime.py

```diff
@@ -321,6 +321,11 @@ def load_dependencies(self, base_only=False) -> Mapping[str, "RuntimeConfig"]:
         if self.dependencies is None:
             all_projects = {self.project_name: self}
             internal_packages = get_include_paths(self.credentials.type)
+            if self.python_adapter_credentials:
+                internal_packages += get_include_paths(self.python_adapter_credentials.type)
+                # remove any repeated project (global project)
+                internal_packages = list(set(internal_packages))
```

Comment:
ATL: Is the order important for the `internal_packages`? Since making it a set would loose the ordering. (If not, maybe let's add a comment that the order doesn't matter).
```
>>> a = ['postgres/macros', 'databricks_macros', 'global/macros', 'python/macros', 'global/macros']
>>> print(list(set(a)))
['global/macros', 'databricks_macros', 'postgres/macros', 'python/macros']
```

Reply:
No, it is then converted into a dict: https://github.com/fal-ai/dbt-core/blob/010ea3e403b75df8a6376aab9ac2df2dcc5f7cf2/core/dbt/config/runtime.py#L345-L353